src/common/database.c

```diff
@@ -1030,20 +1023,21 @@ dt_database_t *dt_database_init(const char *alternative)
             fprintf(
                 stderr,
                 "[init] the database lock file contains a pid that seems to be alive in your system: %d\n",
-                other_pid);
+                pid);
           }
-        }
-        else
-        {
-          fprintf(stderr, "[init] the database lock file seems to be empty\n");
-        }
-        close(fd);
-      }
-      else
-      {
-        fprintf(stderr, "[init] error opening the database lock file for reading\n");
-      }
     }
+    }
+    else  // No lockfile exists - write our PID into new oneit
+    {
+        fd = fopen(db->lockfile, "w");
+        if (!fd) {
+            fprintf(stderr, "[init] error opening the database lock file for writing\n");            
```

Comment:
Yeah this is definitely wrong, if this triggers we would print something to stdout and then try the next line doing fprintf(NULL, ...)